apps/docs/pages/guides/auth/auth-postgres-hooks.mdx

```diff
@@ -0,0 +1,669 @@
+
+### Inputs
+
+- `user_id`Unique identifier for the user attempting to sign in. Correlate this to the `auth.users` table.
+- `claims` Claims which are attached to the access token.
+- `authentication_method` the authentication method used to request for the access token. Possible values include:
+    - `oauth`
+    - `password`
+    - `otp`
+    - `totp`
+    - `recovery`
+    - `invite`
+    - `sso/saml`
+    - `magiclink`
+    - `email/signup`
+    - `email_change`
+    - `token_refresh`
+
+### Outputs
+
+In exchange your hook should return back a JSON object with these fields set:
+
+********************************On Success********************************
+
+Return these only if your hook processed the input without errors.
+
+- `claims` A json containing the updated claims after the hook has been run
+
+****************On Error****************
+
+Return this if your hook processed the input with some error.
+
+- `error` An object that contains information about the error.
+    - `http_code` A number indicating the HTTP code to be returned. If not set, the code is HTTP 500 Internal Server Error.
+    - `message` A message to be returned in the HTTP response. Must be set.
+- In general all other fields attached to the returned JSON are ignored.
+
+
+
+<Admonition type="warn">
+Claims returned must conform to our specification. While you are free to add claims, you can only remove the `user_metadata` and app_metadata claims as we rely on the rest  of the claims for proper functioning of the service.
+</Admonition>
+Example output:
+
+```jsx
+{
+  "error": {
+    "http_code": 429,
+    "message": "You can only verify a factor once very 10 seconds."
+  }
+}
+```
+
+## Examples
+
+<Accordion
+ type="default"
+  openBehaviour="multiple"
+  chevronAlign="right"
+  justified
+  size="medium"
+  className="text-foreground-light">
+<div className="border-b pb-3">
+### Example: Limit Failed MFA Verification Attempts
+<Accordion.Item
+        header={<span className="text-foreground font-bold">Details </span>}
+        id={`limit-failed-mfa-verification-attempts`}
+      >
+
+****Scenario****
+
+You wish to limit the number of failed MFA verification attempts a user can have in an amount of time. A user can only input an incorrect code every 2 seconds and not more than that.
+
+****************************Database Setup****************************
+
+First you need to create a table that will record the last time a user had an incorrect MFA verification attempt for a factor.
+
+You can do this by creating a table such as:
+
+```sql
+create table public.mfa_failed_verification_attempts (
+  user_id uuid not null,
+  factor_id uuid not null,
+  last_failed_at timestamp not null default now(),
+  primary key (user_id, factor_id)
+);
+```
+
+Then you should create a hook which will read and enter information into this table. Example:
+
+```sql
```

Comment:
For all SQL code snippets, try to use `  ` tabs to cut down on visual noise on the page.